Source/CMakeLists.txt

```diff
@@ -21,7 +21,7 @@ add_definitions(-D__STDC_CONSTANT_MACROS)
 
 macro(add_dolphin_library lib srcs libs)
 	add_library(${lib} STATIC ${srcs})
-	target_link_libraries(${lib} ${libs})
+	target_link_libraries(${lib} PUBLIC ${libs})
```

Comment:
The other changes in this commit are good, but I don't agree with this one. It should be PRIVATE by default.